executors_test.go

```diff
@@ -1470,6 +1489,44 @@ func Test_Update_UUID(t *testing.T) {
 	})
 }
 
+func Test_Update_With_Non_ID_PK(t *testing.T) {
+	if PDB == nil {
+		t.Skip("skipping integration tests")
+	}
+	transaction(func(tx *Connection) {
+		r := require.New(t)
+
+		cc := CrookedColour{
+			Name: "You?",
+		}
+		err := tx.Create(&cc)
+		r.NoError(err)
+
+		cc.Name = "Me!"
+		r.NoError(tx.Update(&cc))
```

Comment:
You should check for PK values both after create and after update. I would also do a `p.Connection(ctx).Find(&cl, id)` / `Reload` statement before updating to be sure that getting also works as expected.